README.md

```diff
@@ -446,6 +446,8 @@ The contents of a `.nvmrc` file **must** be the `<version>` (as described by `nv
 
 You can use [`avn`](https://github.com/wbyoung/avn) to deeply integrate into your shell and automatically invoke `nvm` when changing directories. `avn` is **not** supported by the `nvm` development team. Please [report issues to the `avn` team](https://github.com/wbyoung/avn/issues/new).
 
+You can also use [`nvshim`](https://github.com/iamogbz/nvshim) to shim `node` and automatically use the `.nvmrc` file in the current directory. `nvshim` is **not** supported by the `nvm` development team. Please [report issues to the `nvshim` team](https://github.com/iamogbz/nvshim/issues/new).
```

Comment:
Does this also shim `npm` and `npx`? What about available global modules?

Reply:
It shims `node`, `npm` and `npx` https://github.com/iamogbz/nvshim/blob/2b1781f6020c41638194f7fe33e7188be30ca16d/src/nvshim/utils/constants.py#L11

Since it's using the version of node specified by `.nvmrc` it has access to those global modules, same as if you had done `nvm use`.

Reply:
What I mean is, let's say i had `npm install -g eslint` in that node version. Before actually running `nvm use`, how would i be able to run `eslint`?

Reply:
Depends, since this is just removing the need for `nvm use` a couple scenarios apply
* You already are using the nvshim - `npm install -g eslint` would install `eslint` using the version of npm specified by the `.nvmrc` config to npm `NODE_PATH` and this would be accessible by just calling `eslint` if it's in your path or calling `npx eslint` from the folder with the same `.nvmrc` file
  * Note the shim hijacks all calls to node, npm and npx so it would not matter you didn't call `nvm use` before running the install command
* You were not using/installed the nvshim - the install command would install `eslint` using the global npm you have configured as usual, then afterwards `nvshim` is installed, if the version of node in the `.nvmrc` file is not the same as the global version used to install node then it wouldn't have access to the global package via `npx eslint` since `npx` is now shimmed, but `eslint` might still be directly accessible if the install path is in your `PATH`

Hope that clears it up, feel free too try it out yourself. It's pretty easy to install and uninstall, should not leave any annoying artifacts if you choose not to keep it.

Reply:
I see how that would work when i'm *installing* it. The question is, let's say I installed it yesterday, and today started using nvshim for that node version, and ran `eslint` by itself - no node/npm/npx commands whatsoever. What would happen?

Reply:
Not quite sure the edge case you're trying to highlight, but it should just work. That is, it would execute the `eslint` bin file (global) with the `node` bin currently in the `PATH`, which after using `nvshim` would be the shimmed `node` bin.

Shell logs from an attempt to recreate the example you gave in a vm

```sh
gitpod /workspace/oss-boilerplate $ cd ~
gitpod ~ $ pwd
/home/gitpod
gitpod ~ $ which npm
/home/gitpod/.nvm/versions/node/v14.16.0/bin/npm
gitpod ~ $ npm i -g eslint
/home/gitpod/.nvm/versions/node/v14.16.0/bin/eslint -> /home/gitpod/.nvm/versions/node/v14.16.0/lib/node_modules/eslint/bin/eslint.js
+ eslint@7.23.0
added 126 packages from 74 contributors in 6.564s
gitpod ~ $
gitpod ~ $ which eslint
/home/gitpod/.nvm/versions/node/v14.16.0/bin/eslint
gitpod ~ $ cd -
/workspace/oss-boilerplate
gitpod /workspace/oss-boilerplate $ echo 'v10.24.0' > .nvmrc
gitpod /workspace/oss-boilerplate $ pip install nvshim
Collecting nvshim
  Downloading nvshim-1.0.5-py3-none-any.whl (12 kB)
Collecting semver
  Downloading semver-2.13.0-py2.py3-none-any.whl (12 kB)
Collecting colored
  Downloading colored-1.4.2.tar.gz (56 kB)
     |████████████████████████████████| 56 kB 2.7 MB/s 
Building wheels for collected packages: colored
  Building wheel for colored (setup.py) ... done
  Created wheel for colored: filename=colored-1.4.2-py3-none-any.whl size=14003 sha256=d7a4e386d9c1b764cd721be56af263cd212593236f848cbe9bb2863d8a25962e
  Stored in directory: /home/gitpod/.cache/pip/wheels/ae/61/d2/51d0185d88a932058c7341ae08278b27f5312f3667efe433fc
Successfully built colored
Installing collected packages: semver, colored, nvshim
Successfully installed colored-1.4.2 nvshim-1.0.5 semver-2.13.0
gitpod /workspace/oss-boilerplate $
gitpod /workspace/oss-boilerplate $ which node
/workspace/.pip-modules/bin/node
gitpod /workspace/oss-boilerplate $ which eslint
/home/gitpod/.nvm/versions/node/v14.16.0/bin/eslint
gitpod /workspace/oss-boilerplate $ eslint --version
N/A version 'v10.24.0' is not yet installed.

You need to run 'nvm install v10.24.0' to install it before using it.

Or set the environment variable 'NVSHIM_AUTO_INSTALL' to auto install at run time.

gitpod /workspace/oss-boilerplate $ export NVSHIM_AUTO_INSTALL=1
gitpod /workspace/oss-boilerplate $ eslint --version
Downloading and installing node v10.24.0...
Downloading https://nodejs.org/dist/v10.24.0/node-v10.24.0-linux-x64.tar.xz...
############################################################################################################################################ 100.0%
Computing checksum with sha256sum
Checksums matched!
Now using node v10.24.0 (npm v6.14.11)
v7.23.0
gitpod /workspace/oss-boilerplate $ 
```

Reply:
What would `which eslint` report, on a new shell, before i'd ever invoked `node` or `npm` or `npx`?

Reply:
Depends on what you have in your `PATH`, but it should be the the global installed `eslint`. Okay so `nvshim` does not modify your `PATH` in anyway...

Reply:
Right - the fundamental flaw with any "shimming" approach is exactly this. Let's say node v14 has `eslint` installed, and  node v15 has `prettier` installed. On a brand new shell, I should be able to invoke `eslint` (if node 14 is the one that would be selected) or `prettier` (if node 15 is the one that would be selected).

If I can't do that, then it's not a transparent shim, and users would need to know the caveats of using it.

Reply:
Hmm, so if I understand correctly, you'd expect all bins installable using the shimmed `npm` to also be shimmed? I guess I can make it clearer that only the `node` binaries are shimmed i.e. node, npm and npx. I think that's transparent and a reasonable scope since node provides a default way to execute bins using `npx`.

Also I think nvm already covers the brand new shell case, not so much when you change directories though.

Reply:
Yes, that's what I'd expect - that a tool that avoids the need to run `nvm use X` would also have all globally installed modules within node X available as well.

If the caveats are unavoidable, then to get this merged, I'd like both the repo's docs and the paragraph here to very explicitly note these caveats.